contracts/extensions/MultisigPermissions.sol

```diff
@@ -0,0 +1,590 @@
+
+// ignore-file-swc-108
+
+contract MultisigPermissions is ColonyExtensionMeta, ColonyDataTypes, GetActionSummary {
+  using Bytes4Includes for bytes4[];
+
+  // Events
+
+  event MultisigRoleSet(address agent, address user, uint256 domainId, uint256 roleId, bool setTo);
+  event MotionExecuted(address agent, uint256 motionId, bool success);
+  event MotionCancelled(address agent, uint256 motionId);
+  event MotionCreated(address agent, uint256 motionId);
+  event ApprovalChanged(address agent, uint256 motionId, uint8 role, bool approval);
+  event RejectionChanged(address agent, uint256 motionId, uint8 role, bool approval);
+  event GlobalThresholdSet(uint256 globalThreshold);
+  event DomainSkillThresholdSet(uint256 domainSkillId, uint256 threshold);
+
+  // Datatypes
+
+  struct Motion {
+    address[] targets;
+    bytes[] data;
+    uint256 approvalCount;
+    uint256 rejectionCount;
+    uint256 domainSkillId;
+    bytes32 requiredPermissions;
+    uint256 overallApprovalTimestamp;
+    uint256 creationTimestamp;
+    address creator;
+    bool executed;
+    bool rejected;
+    bytes4 sig;
+  }
+
+  enum Vote {
+    None,
+    Approve,
+    Reject
+  }
+
+  // Constants
+  bytes4 constant OLD_MOVE_FUNDS =
+    bytes4(
+      keccak256("moveFundsBetweenPots(uint256,uint256,uint256,uint256,uint256,uint256,address)")
+    );
+
+  // Storage
+
+  // User Address => Domain Id => Roles
+  mapping(address => mapping(uint256 => bytes32)) internal userDomainRoles;
+
+  IColonyNetwork colonyNetwork;
+  uint256 globalThreshold;
+
+  // Domain Skill Id => Role => Usercount
+  mapping(uint256 => mapping(uint8 => uint256)) domainSkillRoleCounts;
+  // Domain Skill Id => Fixed Threshold
+  mapping(uint256 => uint256) domainSkillThreshold;
+
+  uint256 motionCount;
+  mapping(uint256 => Motion) motions;
+
+  // Motion Id => User => Vote => Permissions
+  mapping(uint256 => mapping(address => mapping(Vote => bytes32))) motionVotes;
+  // Motion Id => Vote => Role => Vote Count
+  mapping(uint256 => mapping(Vote => mapping(uint8 => uint256))) motionVoteCount;
+
+  // Overrides
+
+  /// @notice Returns the identifier of the extension
+  /// @return _identifier The extension's identifier
+  function identifier() public pure override returns (bytes32 _identifier) {
+    return keccak256("MultisigPermissions");
+  }
+
+  /// @notice Returns the version of the extension
+  /// @return _version The extension's version number
+  function version() public pure override returns (uint256 _version) {
+    return 1;
+  }
+
+  /// @notice Configures the extension
+  /// @param _colony The colony in which the extension holds permissions
+  function install(address _colony) public override auth {
+    super.install(_colony);
+    colonyNetwork = IColonyNetwork(colony.getColonyNetwork());
+  }
+
+  function getCapabilityRoles(bytes4 _sig) public pure override returns (bytes32) {
+    if (_sig == SET_USER_ROLES) {
```

Comment:
Does this need updating to account for the other role-change functions?

Reply:
No, because they're not functions on this contract.